### PR TITLE
Updated serialisation system to use function pointers

### PR DESCRIPTION
**Pull request by J-Mo63**
See changes. Let me know if they're ok and I can do cleanup.

### CODE REVIEW

File: src/systems/resource/ResourceManager.h [hunk 1]
```diff
@@ -6,9 +6,9 @@
 #include <variant>
 #include <vector>
 
-// Define macros
-#define RESOURCE_MAP std::map<std::string, std::variant<Model, Texture2D>>
-#define RESOURCE_LIST std::vector<std::variant<Model, Texture2D>*>
```
Comment (Raelr) [inline src/systems/resource/ResourceManager.h:11]: I guess I did define some macros... I feel like the `ResourceManager` is going to need a bit of love in future. For now it does exactly what we need it to.

File: src/systems/scene/SceneSerialiser.h [hunk 2]
```diff
@@ -7,6 +7,15 @@
 #include <map>
 #include <Vector3.hpp>
 
+// Define constants
+static constexpr const char SEP = '|';
+static constexpr const char NAME_SEP = ':';
+
+// Define types
+typedef std::string(Serialiser)(class Entity* entity);
```
Comment (Raelr) [inline src/systems/scene/SceneSerialiser.h:15]: I'm finding this to be incredibly confusing to read. Could you comment the first two `typedef`s here? I'm not 100% sure I understand what these are doing on a syntactical level.

**Review by Raelr — COMMENTED** [review]
Nicely done. I don't have too many comments on the overall syntax. To me, there's just some syntax that's not entirely transparent on what it does, however this could be due to my own lack of understanding of C/C++ syntax.

File: src/systems/scene/SceneSerialiser.h (hunk 2 above)
Comment (J-Mo63) [reply src/systems/scene/SceneSerialiser.h:15]: Replaced them with `std::function`s should be more readable now.

**Comment by J-Mo63** [comment]
> To me, there's just some syntax that's not entirely transparent on what it does, however this could be due to my own lack of understanding of C/C++ syntax.

Ok well then you're really not going to like what I've just done, but it sure is a useful thing.

**Review by Raelr — APPROVED** [review]
Just saw your changes. They look alright to me, the new typedefs are a lot easier to understand and the macros aren't TOO bad to read. LGTM 👍

**Pull request merged**